ledger/sandbox-on-x/src/test/it/scala/com/daml/ledger/platform/sandbox/auth/AudienceBasedTokenAuthIT.scala

```diff
@@ -0,0 +1,132 @@
+    Some(
+      toHeader(
+        payload,
+        audienceBasedToken = true,
+      )
+    )
+  )
+
+  val expectedAudienceToken: StandardJWTPayload = StandardJWTPayload(
+    issuer = None,
+    participantId = None,
+    userId = "participant_admin",
+    exp = None,
+    format = StandardJWTTokenFormat.ParticipantId,
+    audiences = List(ExpectedAudience),
+  )
+
+  val multipleAudienceWithExpectedToken: StandardJWTPayload =
+    expectedAudienceToken.copy(audiences = List(ExpectedAudience, "additionalAud"))
+
+  val noAudienceWithExpectedToken: StandardJWTPayload =
+    expectedAudienceToken.copy(audiences = List())
+
+  val wrongAudienceWithExpectedToken: StandardJWTPayload =
+    expectedAudienceToken.copy(audiences = List("aud1", "aud2"))
+
+  val expiredToken: StandardJWTPayload =
+    expectedAudienceToken.copy(exp = Some(Instant.now().plusNanos(Duration.ofDays(-1).toNanos)))
+
+  val unknownUserToken: StandardJWTPayload =
+    expectedAudienceToken.copy(userId = "unknown_user")
+
+  val scopeToken = expectedAudienceToken.copy(format = StandardJWTTokenFormat.ParticipantId)
+
+  it should "allow access to an endpoint with the token which is matching intended audience" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can make a call with a JWT with intended audience"
+    ) in {
+    expectSuccess {
+      serviceCall(toContext(expectedAudienceToken))
+    }
+  }
+
+  it should "allow access to an endpoint with the token with multiple audiences which is matching expected audience" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can make a call with a JWT with intended audience"
+    ) in {
+    expectSuccess {
+      serviceCall(toContext(multipleAudienceWithExpectedToken))
+    }
+  }
+
+  it should "allow access to an endpoint with the token which is matching intended audience and scope defined" taggedAs securityAsset
+    .setHappyCase(
+      "Ledger API client can make a call with a JWT with intended audience"
+    ) in {
+    expectSuccess(serviceCall(toContext(scopeToken)))
+  }
+
+  it should "deny access with no intended audience" taggedAs securityAsset.setAttack(
+    attackUnauthenticated(threat =
+      "Ledger API client cannot make a call with a JWT with no intended audience"
+    )
+  ) in {
+    expectUnauthenticated(serviceCall(toContext(noAudienceWithExpectedToken)))
+  }
+
+  it should "deny access with wrong intended audience" taggedAs securityAsset.setAttack(
+    attackUnauthenticated(threat =
+      "Ledger API client cannot make a call with a JWT with wrong intended audience"
+    )
+  ) in {
+    expectUnauthenticated(serviceCall(toContext(wrongAudienceWithExpectedToken)))
+  }
+
+  it should "deny calls with user token for 'unknown_user' without expiration" taggedAs adminSecurityAsset
+    .setAttack(
+      attackPermissionDenied(threat = "Present a user JWT for 'unknown_user' without expiration")
```

Comment:
What is the deciding factor for this threat to be permission_denied and for others to be unauthenticated?

Reply:
Somehow this is what was there in the impl before. Depending on which layer authorization fails - it gives either permission denied or unauthenticated. Permission denied usually is thrown when something has been correctly provided. "Aud" claim check happens during parsing of the token, so it fails very early which is why it throws "unauthenticated"